lib/jekyll-github-metadata/client.rb

```diff
@@ -71,13 +72,25 @@ def inspect
         "#<#{self.class.name} @client=#{client_inspect}>"
       end
 
+      def authenticated?
+        !@client.access_token.nil? && !@client.access_token.empty?
+      end
+
+      # Raise an error if credentials are provided and they cause a 401.
+      def check_credentials!
+        return unless authenticated?
+        @client.user
```

Comment:
Why make an extra call to check credentials, when we can just rescue if the call we need fails due to bad auth?